app.py

```diff
@@ -1,80 +1,102 @@
-        Experience("Software Developer",
-                   "A Cool Company",
-                   "October 2022",
-                   "Present",
-                   "Writing Python Code",
-                   "example-logo.png")
+        Experience(
+            "Software Developer",
+            "A Cool Company",
+            "October 2022",
+            "Present",
+            "Writing Python Code",
+            "example-logo.png",
+        )
     ],
     "education": [
-        Education("Computer Science",
-                  "University of Tech",
-                  "September 2019",
-                  "July 2022",
-                  "80%",
-                  "example-logo.png")
+        Education(
+            "Computer Science",
+            "University of Tech",
+            "September 2019",
+            "July 2022",
+            "80%",
+            "example-logo.png",
+        )
     ],
-    "skill": [
-        Skill("Python",
-              "1-2 Years",
-              "example-logo.png")
-    ]
+    "skill": [Skill("Python", "1-2 Years", "example-logo.png")],
 }
 
 
-@app.route('/test')
+@app.route("/test")
 def hello_world():
-    '''
+    """
     Returns a JSON test message
-    '''
+    """
     return jsonify({"message": "Hello, World!"})
 
 
-@app.route('/resume/experience', methods=['GET', 'POST'])
+@app.route("/resume/experience", methods=["GET", "POST"])
 def experience():
-    '''
+    """
     Handle experience requests
-    '''
-    if request.method == 'GET':
+    """
+    if request.method == "GET":
         return jsonify()
 
-    if request.method == 'POST':
+    if request.method == "POST":
         return jsonify({})
 
     return jsonify({})
 
-@app.route('/resume/education', methods=['GET', 'POST'])
+
+@app.route("/resume/education", methods=["GET", "POST"])
 def education():
-    '''
+    """
     Handles education requests
-    '''
-    if request.method == 'GET':
+    """
+    if request.method == "GET":
         return jsonify({})
 
-    if request.method == 'POST':
+    if request.method == "POST":
         return jsonify({})
 
     return jsonify({})
 
 
-@app.route('/resume/skill', methods=['GET', 'POST'])
+@app.route("/resume/skill", methods=["GET", "POST"])
 def skill():
-    '''
+    """
     Handles Skill requests
-    '''
-    if request.method == 'GET':
+    """
+    if request.method == "GET":
         return jsonify({})
 
-    if request.method == 'POST':
-        return jsonify({})
+    if request.method == "POST":
+        json_data = request.json
+        try:
+            # extract the data from the request
+            name = json_data["name"]
+            proficiency = json_data["proficiency"]
+            logo = json_data["logo"]
+
+            new_skill = Skill(name, proficiency, logo)
+
+            data["skill"].append(new_skill)
+
+            # return the new skill and its ID
+            return jsonify(
+                {"id": len(data["skill"]) - 1, "skill": new_skill.__dict__}
```

Comment:
Do we have to return both the new skill and the index. IIRC the bug just asks to return the index

Reply:
Yes, they stated just to return the index. That was just an addition on my end, I'll rectify it now